lib/srv/db/proxyserver.go

```diff
@@ -250,7 +250,8 @@ func (s *ProxyServer) handleConnection(conn net.Conn) error {
 func (s *ProxyServer) dispatch(clientConn net.Conn) (common.Proxy, error) {
 	muxConn, ok := clientConn.(*multiplexer.Conn)
 	if !ok {
-		return nil, trace.BadParameter("expected multiplexer connection, got %T", clientConn)
+		// Postgres was configured on a separate proxy listeners.
+		return s.PostgresProxy(), nil
 	}
 	switch muxConn.Protocol() {
 	case multiplexer.ProtoPostgres:
```

Comment:
Seems like this function handles only Postgres connections (separate listener or not) anyway, is there any point in still having it? Let's get rid of it and just use `s.PostgresProxy()` directly in where all Postgres connections are handled?